source/common/quic/quic_transport_socket_factory.cc

```diff
@@ -16,8 +16,9 @@ QuicServerTransportSocketConfigFactory::createTransportSocketFactory(
       config, context.messageValidationVisitor());
   auto server_config = std::make_unique<Extensions::TransportSockets::Tls::ServerContextConfigImpl>(
       quic_transport.downstream_tls_context(), context);
-  auto factory =
-      std::make_unique<QuicServerTransportSocketFactory>(context.scope(), std::move(server_config));
+  auto factory = std::make_unique<QuicServerTransportSocketFactory>(
+      (quic_transport.has_enable_early_data() ? quic_transport.enable_early_data().value() : true),
+      context.scope(), std::move(server_config));
```

Comment:
Could we have a unit test of createTransportSocketFactory which verifies the expected value of the returned factory's enable_early_data in which we pass in quic_transport with no value, true and false for enable_early_data?

Reply:
done